src/Deserializers/AliasGroupDeserializer.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Wikibase\DataModel\Deserializers;
+
+use Deserializers\Deserializer;
+use Deserializers\Exceptions\DeserializationException;
+use Deserializers\Exceptions\InvalidAttributeException;
+use Deserializers\Exceptions\MissingAttributeException;
+use Wikibase\DataModel\Term\AliasGroup;
+
+/**
+ * @since 1.5
+ *
+ * @author Adam Shorland
+ */
+class AliasGroupDeserializer implements Deserializer {
+
+	public function __construct() {
+	}
+
+	/**
+	 * @param mixed $serialization
+	 *
+	 * @return AliasGroup
+	 * @throws DeserializationException
+	 */
+	public function deserialize( $serialization ) {
+		$this->assertCanDeserialize( $serialization );
+
+		return $this->getDeserialized( $serialization );
+	}
+
+	/**
+	 * @param array $serialization
+	 *
+	 * @return AliasGroup
+	 */
+	private function getDeserialized( $serialization ) {
+		foreach ( $serialization as $requestedLanguage => $aliasesPerLanguageSerialization ) {
+			if ( !is_array( $aliasesPerLanguageSerialization ) ) {
+				throw new DeserializationException( "Aliases serialization should be an array of array" );
+			}
+
+			$aliases = array();
+
+			foreach ( $aliasesPerLanguageSerialization as $aliasSerialization ) {
+				$this->assertIsValidValueSerialization( $aliasSerialization, $requestedLanguage );
+				$aliases[] = $aliasSerialization['value'];
+			}
+
+			return new AliasGroup(
+				$requestedLanguage,
+				$aliases
+			);
+		}
+	}
+
+	/**
+	 * @param array $serialization
+	 */
+	private function assertCanDeserialize( $serialization ) {
+		if ( !is_array( $serialization ) ) {
+			throw new DeserializationException( 'The alias group serialization should be an array' );
+		}
+		if( count( $serialization ) > 1 ) {
+			throw new DeserializationException( 'The alias group serialization should only contain 1 element' );
+		}
+	}
+
+	private function assertIsValidValueSerialization( $serialization, $requestedLanguage ) {
+		if ( !is_array( $serialization ) ) {
+			throw new DeserializationException( 'Term serializations must be arrays' );
+		}
+
+		$this->requireAttribute( $serialization, 'language' );
+		$this->requireAttribute( $serialization, 'value' );
+		$this->assertNotAttribute( $serialization, 'source' );
+
+		$this->assertAttributeInternalType( $serialization, 'language', 'string' );
+		$this->assertAttributeInternalType( $serialization, 'value', 'string' );
+		$this->assertRequestedAndActualLanguageMatch( $serialization, $requestedLanguage );
+	}
+
+	private function requireAttribute( array $array, $attributeName ) {
+		if ( !array_key_exists( $attributeName, $array ) ) {
```

Comment:
We generally use `array_key_exists`. AFAIK we have no real rules about it, though personally I strongly recommend using that as it is more intent revealing.